### PR TITLE
Refs #28643 -- Reorganized database functions.

### PR DESCRIPTION
**Pull request by felixxm**

### CODE REVIEW

File: docs/ref/models/database-functions.txt [hunk 1]
```diff
@@ -175,8 +144,67 @@ will result in a database error.
...
+    current transaction started. Therefore for cross-database compatibility,
+    ``Now()`` uses ``STATEMENT_TIMESTAMP`` instead. If you need the transaction
+    timestamp, use :class:`django.contrib.postgres.functions.TransactionNow`.
+
+.. _text-functions:
+
+Text Functions
+==============
+
+.. module:: django.db.models.functions.text
```
Comment (timgraham) [inline docs/ref/models/database-functions.txt]: I think we shouldn't document the functions in the "text" namespace. Most other references in Django are documented in the convenient import location. Not doing that for functions.datetime was probably a mistake.

**Comment by felixxm** [comment]
Jenkins

**Comment by felixxm** [comment]
@timgraham Thanks for suggestions. I totally agree with you. I added plan to the ticket (see [comment](https://code.djangoproject.com/ticket/28643#comment:9)) and reorganized existing docs and code.

**Comment by timgraham** [comment]
Perhaps the "misc" functions should be documented in their own section rather than at the top of the page at a separate heading level from the categorized functions. I'd make the documentation changes separately so they can be backported to stable/2.0.x, otherwise we're going to have pain backporting any documentation changes to that file in the coming months.

**Comment by felixxm** [comment]
Jenkins.

**Comment by felixxm** [comment]
@timgraham I prepared separate PR with docs changes #9234.

File: django/db/models/functions/comp_conv.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
+"""
+Classes that represent database functions.
```
Comment (timgraham) [inline django/db/models/functions/comp_conv.py]: `"""Database functions that do comparisons or type conversions."""`

File: django/db/models/functions/__init__.py [hunk 3]
```diff
@@ -1,27 +1,27 @@
-from .base import (
-    Cast, Coalesce, Concat, ConcatPair, Greatest, Least, Length, Lower, Now,
-    StrIndex, Substr, Upper,
-)
+from .comp_conv import Cast, Coalesce, Greatest, Least
```
Comment (timgraham) [inline django/db/models/functions/__init__.py]: I think "comparison.py" is fine.

File: django/db/models/functions/datetime.py [hunk 4]
```diff
@@ -143,6 +143,17 @@ class ExtractSecond(Extract):
 ExtractYear.register_lookup(YearLte)
 
 
+class Now(Func):
+    template = 'CURRENT_TIMESTAMP'
+    output_field = fields.DateTimeField()
+
+    def as_postgresql(self, compiler, connection):
+        # Postgres' CURRENT_TIMESTAMP means "the time at the start of the
+        # transaction". We use STATEMENT_TIMESTAMP to be cross-compatible with
```
Comment (timgraham) [inline django/db/models/functions/datetime.py]: While moving this, chop "we".

File: django/db/models/functions/datetime.py [hunk 5]
```diff
@@ -143,6 +143,17 @@ class ExtractSecond(Extract):
 ExtractYear.register_lookup(YearLte)
 
 
+class Now(Func):
+    template = 'CURRENT_TIMESTAMP'
+    output_field = fields.DateTimeField()
+
+    def as_postgresql(self, compiler, connection):
+        # Postgres' CURRENT_TIMESTAMP means "the time at the start of the
```
Comment (timgraham) [inline django/db/models/functions/datetime.py]: "Postgres'" looks odd to me. Use "PostgreSQL's" instead.

File: django/db/models/functions/comp_conv.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    """Return, from left to right, the first non-null expression."""
+    function = 'COALESCE'
+
+    def __init__(self, *expressions, **extra):
+        if len(expressions) < 2:
+            raise ValueError('Coalesce must take at least two expressions')
+        super().__init__(*expressions, **extra)
+
+    def as_oracle(self, compiler, connection):
+        # we can't mix TextField (NCLOB) and CharField (NVARCHAR), so convert
```
Comment (timgraham) [inline django/db/models/functions/comp_conv.py]: rewrite without "we": "Oracle prohibits mixing TextField (NCLOB) and CharField (NVARCHAR), so convert all fields to  NCLOB when that type is expected."

File: django/db/models/functions/comp_conv.py [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        # all fields to NCLOB when we expect NCLOB
+        if self.output_field.get_internal_type() == 'TextField':
+            class ToNCLOB(Func):
+                function = 'TO_NCLOB'
+
+            expressions = [
+                ToNCLOB(expression) for expression in self.get_source_expressions()]
```
Comment (timgraham) [inline django/db/models/functions/comp_conv.py]: put closing ] on next line

**Pull request merged**